test/cmd_line/substitute.test.ts

```diff
@@ -303,6 +303,106 @@ suite('Basic substitute', () => {
 
       assertEqualLines(['fighters', 'bar', 'fighters', 'bar']);
     });
+    test('Substitute with parameters should update search state', async () => {
+      await modeHandler.handleMultipleKeyEvents([
+        'i', // foo bar foo bar
```

Comment:
You might take a look at `test/testSimplifier.js`, such as  `asVimInputText` and `tokenizeKeySequence`. Most of our standard tests use the helpers there to input text in a concise, readable manner. This doesn't have to use the helper, but it might with this call at least.

It could look something like this:

```
  const enterKeys = keys => modeHandler.handleMultipleKeyEvents(tokenizeKeySequence(keys))
  await enterKeys(['foo', 'bar', 'foo', 'bar'].join('\n'));
  await enterKeys('/bar');
  await waitForCursorSync();
```

Reply:
Oh, cool!  Yeah, that's way easier on the eyes!  Thanks!

Sorry about not seeing that...  I only looked at the tests nearby and only saw this pattern.

Will put it on my list of things to change for the next commit. :)

Reply:
- [ ] rewrite tests using awesome helpers so they're easier on people's eyes :P

Reply:
@shawnaxsom I used newTest, and you were right - it's way better :)

Feel free to let me know if there's anything else.